File: Basic/Python/QiPy/QiNamespace.py
import json

class QiNamespace(object):
    """description of class"""

    def __init__(self):
        self.__Id = 0
    
    def getId(self):
        return self.__Id

    def setId(self, id):
        self.__Id = id

    Id = property(getId, setId)

    def toString(self):
        return json.dumps(self.toDictionary())

    def toDictionary(self):
        
        dictionary = { "Id" : self.__Id }

        return dictionary

    @staticmethod
    def fromString(content):
         dictionary = json.loads(content)
         return QiNamespace.fromDictionary(dictionary)

    @staticmethod
    def fromDictionary(content):
        namespace = QiNamespace()

        if len(content) == 0:
            return typeProperty

        if "Id" in content:
            namespace.Id = content["Id"]
            
        return namespace

File: Basic/Java/src/samples/QiNamespace.java
package samples;

public class QiNamespace 
{
    private String Id;
    
    public QiNamespace() 
    {
        this.Id = "Default";
    }
    
    public QiNamespace(String namespaceId) 
    {
        this.Id = namespaceId;
    }
    
    public String getId() 
    {
        return this.Id;
    }
    
    public void setId(String namespaceId) 
    {
        this.Id = namespaceId;
    }
}
